neoutils/connection.go

```diff
@@ -50,6 +52,12 @@ func DefaultConnectionConfig() *ConnectionConfig {
 	}
 }
 
+//We need to initialize the logger as this library relies on an initialized global variable for the logger object. 
+//Otherwise dependent services that don't initialize this global object get errors using the library.
+func init() {
+	logger.InitDefaultLogger("neo4j-utils-go")
```

Comment:
As we don't have guaranties for the order of `init` functions executions and there are services that use `init` for initialization of the logger I would try to make sure that the logger is not initialized already.
```suggestion
	entry := &logrus.Entry{}
	_, err := logger.Logger().Formatter.Format(entry)
	if err != nil && err.Error() == "logger is not initialised - please use InitLogger or InitDefaultLogger function" {
		logger.InitDefaultLogger("neo4j-utils-go")
	}
```

Reply:
I tested it and this fix won't help as this logger is initialized before the logger of the dependent service.